src/newgrf_gui.cpp

```diff
@@ -1095,6 +1101,8 @@ struct NewGRFWindow : public Window, NewGRFScanCallback {
 					CopyGRFConfigList(this->orig_list, this->actives, true);
 					ResetGRFConfig(false);
 					ReloadNewGRFData();
+					modified = false;
+					this->InvalidateData(GOID_NEWGRF_CHANGES_APPLIED);
```

Comment:
`GOID_NEWGRF_CHANGES_APPLIED` already sets modified to false, doesn't it? Also, missing `this->` otherwise.